libs/application/templates/estate/src/fields/EstateMembersRepeater/AdditionalEstateMember.tsx

```diff
@@ -0,0 +1,217 @@
+import { YES } from '../../lib/constants'
+import { IDENTITY_QUERY } from '../../graphql'
+import { hasYes } from '../../lib/utils'
+
+export const AdditionalEstateMember = ({
+  field,
+  index,
+  remove,
+  fieldName,
+  relationOptions,
+  error,
+}: {
+  field: Partial<ArrayField<any, 'id'>>
+  index: number
+  remove: (index?: number | number[] | undefined) => void
+  fieldName: string
+  relationOptions: { value: string; label: string }[]
+  error: any
+}) => {
+  const { formatMessage } = useLocale()
+  const fieldIndex = `${fieldName}[${index}]`
+  const nameField = `${fieldIndex}.name`
+  const nationalIdField = `${fieldIndex}.nationalId`
+  const relationField = `${fieldIndex}.relation`
+  const dateOfBirthField = `${fieldIndex}.dateOfBirth`
+  const foreignCitizenshipField = `${fieldIndex}.foreignCitizenship`
+  const initialField = `${fieldIndex}.initial`
+  const dummyField = `${fieldIndex}.dummy`
+  const enabledField = `${fieldIndex}.enabled`
+  const nationalIdInput = useWatch({ name: nationalIdField, defaultValue: '' })
+  const name = useWatch({ name: nameField, defaultValue: '' })
+  const foreignCitizenship = useWatch({
+    name: foreignCitizenshipField,
+    defaultValue: hasYes(field.foreignCitizenship) ? [YES] : '',
+  })
+
+  const { control, setValue } = useFormContext()
+
+  const [
+    getIdentity,
+    { loading: queryLoading, error: queryError },
+  ] = useLazyQuery<Query, { input: IdentityInput }>(IDENTITY_QUERY, {
+    onCompleted: (data) => {
+      setValue(nameField, data.identity?.name ?? '')
+    },
+    fetchPolicy: 'network-only',
+  })
+
+  useEffect(() => {
+    if (nationalIdInput.length === 10 && kennitala.isValid(nationalIdInput)) {
+      getIdentity({
+        variables: {
+          input: {
+            nationalId: nationalIdInput,
+          },
+        },
+      })
+    } else if (
+      name !== '' &&
+      (!foreignCitizenship || foreignCitizenship.length === 0)
+    ) {
+      setValue(nameField, '')
+    }
+  }, [getIdentity, name, nameField, nationalIdInput, setValue])
+
+  return (
+    <Box position="relative" key={field.id} marginTop={2}>
+      <Controller
+        name={initialField}
+        control={control}
+        defaultValue={field.initial || false}
+      />
+      <Controller
+        name={dummyField}
+        control={control}
+        defaultValue={field.dummy || false}
+      />
+      <Controller
+        name={enabledField}
+        control={control}
+        defaultValue={field.enabled || false}
+      />
+      <Box position="absolute" className={styles.removeFieldButton}>
+        <Button
+          variant="ghost"
+          size="small"
+          circle
+          icon="remove"
+          onClick={() => {
+            remove(index)
+          }}
+        />
+      </Box>
+      <GridRow>
+        {foreignCitizenship[0] !== YES ? (
+          <>
+            <GridColumn span={['1/1', '1/2']} paddingBottom={2}>
+              <InputController
```

Comment:
You could instead of making controllers from scratch use the ones we already have for such as TextFormField by spreading the field property down into the TextFormField and overwriting the needed props.
For example if a TextFormField:
```
        <TextFormField
            application={application}
            error={error}
            showFieldName={true}
            field={{
              ...field,
              type: FieldTypes.TEXT,
              component: FieldComponents.TEXT,
              children: undefined,
              defaultValue: field.nationalId || '',
              id: nationalIdField,
              title: '',
              placeholder: '',
              backgroundColor: 'blue',
              required: true,
              onChange,
            }}
          />
```
That way the input field isnt on its own and updates with changes to the TextFormField so there would be no need to make adjustments if something changes like accessibility.

Just an idea.

Reply:
Hey that's useful. We'll look into it thank you :)

Reply:
the trick with this one is maybe the 'loading' part of it? can i use it with TextFormField? but anyways, we will fix the other fields to use TextFormField :)